types/src/index.ts

```diff
@@ -310,15 +310,23 @@ export interface EmitMetadata {
   ts?: number;
```

Comment:
possibly the timestamp is a prerequisite for the `greatest` strategy. Let me know if that is the case and I can make the change.